src-java/floodlightrouter-topology/floodlightrouter-storm-topology/src/main/java/org/openkilda/wfm/topology/floodlightrouter/model/RegionMappingAdd.java

```diff
@@ -0,0 +1,29 @@
+/* Copyright 2020 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.floodlightrouter.model;
+
+import org.openkilda.model.SwitchId;
+
+public class RegionMappingAdd extends RegionMappingUpdate {
+    public RegionMappingAdd(SwitchId switchId, String region, boolean readWriteMode) {
+        super(switchId, region, readWriteMode);
+    }
+
+    @Override
+    public void apply(RegionMappingStorage storage) {
+        storage.update(this);
```

Comment:
this inverse of action make things harder to track without any pros

Reply:
This is a visitor pattern. What do you propose to use here?